tests/task_queue_test.go

```diff
@@ -659,16 +549,36 @@ func (s *TaskQueueSuite) TestPerKeyRateLimit_Default_IsEnforcedAcrossThreeKeys()
 		tasksPerKey   = 30
 		buffer        = 3 * time.Second // relax for CI jitter
 	)
-	keys := []string{"A", "B", "C"}
+	fairnessKeysWithWeight := map[string]float32{"A": 1.0, "B": 1.0, "C": 1.0}
 
 	tv := testvars.New(s.T())
 
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 
-	perKeyTimes, _ := s.setupPerKeyRateLimitWorkflow(ctx, tv, keys, tasksPerKey, wholeQueueRPS, perKeyRPS)
+	// Fast refresh so ratelimiter picks up config quickly.
+	s.OverrideDynamicConfig(dynamicconfig.TaskQueueInfoByBuildIdTTL, 1*time.Millisecond)
```

Comment:
Just a couple more thoughts, don't need to fix right now:

Why does this dc affect how the rate limiter picks up config? That should be immediate now with subscriptions?
Either the comment and setting are wrong or something is very unintuitive and should be cleaned up.

Reply:
Good question; I suspect a copy/paste case. I've removed it now, let's see if it still succeeds.